aggregator/src/app/BundleTable.ts

```diff
@@ -109,109 +115,167 @@ function toInsertRawRow(row: InsertRow): InsertRawRow {
 
 function toRawRow(row: Row): RawRow {
   return {
-    ...row,
-    submitError: row.submitError ?? null,
-    bundle: JSON.stringify(bundleToDto(row.bundle)),
+    id: row.id,
+    status: row.status,
+    hash: row.hash,
+    bundle: JSON.stringify(row.bundle),
     eligibleAfter: toUint256Hex(row.eligibleAfter),
     nextEligibilityDelay: toUint256Hex(row.nextEligibilityDelay),
+    submitError: row.submitError ?? null,
     receipt: JSON.stringify(row.receipt),
   };
 }
 
 export default class BundleTable {
-  queryTable: QueryTable<RawRow>;
-  safeName: string;
-
-  private constructor(public queryClient: QueryClient, tableName: string) {
-    this.queryTable = this.queryClient.table<RawRow>(tableName);
-    this.safeName = unsketchify(this.queryTable.name);
+  constructor(
+    public db: sqlite.DB,
+    public onQuery = (_sql: string, _params?: sqlite.QueryParameterSet) => {},
+  ) {
+    this.dbQuery(`
+      CREATE TABLE IF NOT EXISTS bundles (
+        id INTEGER PRIMARY KEY AUTOINCREMENT,
+        status TEXT NOT NULL,
+        hash TEXT NOT NULL,
+        bundle TEXT NOT NULL,
+        eligibleAfter TEXT NOT NULL,
+        nextEligibilityDelay TEXT NOT NULL,
+        submitError TEXT,
+        receipt TEXT
+      )
+    `);
   }
 
-  static async create(
-    queryClient: QueryClient,
-    tableName: string,
-  ): Promise<BundleTable> {
-    const table = new BundleTable(queryClient, tableName);
-    await table.queryTable.create(tableOptions, CreateTableMode.IfNotExists);
-
-    return table;
+  dbQuery(sql: string, params?: sqlite.QueryParameterSet) {
+    this.onQuery(sql, params);
+    return this.db.query(sql, params);
```

Comment:
I'm surprised this is synchronous, most file i/o is not. Looks like under the covers it's making calls to a WASM binary. Most sync file i/o in NodeJS (i.e. `fs.readFileSync`) and other JS systems is discouraged as it blocks the event loop. Looks like this might be flushing writes in the background. Worth keeping an eye on if we see any stuttering/performance/persistence issues.

Reply:
Yeah this is an interesting one. I've met this point before, and my understanding is that it's normal and expected to do things synchronously when using sqlite. The idea is that sqlite queries need to be sequential anyway, and they're very fast, so async doesn't really fit. There are exceptions of course, but I don't think they apply to us. [More info](https://github.com/WiseLibs/better-sqlite3#why-should-i-use-this-instead-of-node-sqlite3).